configsuite/extension/ext.py

```diff
@@ -0,0 +1,206 @@
+    "string": "default",
+    "integer": "0",
+    "number": "0.0",
+    "bool": "True",
+    "date": "01.01.1999",
+    "datetime": "01.01.1999",
+}
+
+
+def scb_static_path(app):
+    static_path = os.path.abspath(os.path.join(os.path.dirname(__file__), "_static"))
+    app.config.html_static_path.append(static_path)
+
+
+# we are assuming several keywords.
+# I think we should provide proper user feedback if these are not included.
+def generate(schema, parent_section, parent_title=""):
+    title = schema.get(MK.Description, "default_title")[:10]
+    title = title.lower()
+    title = title.replace(" ", "_")
+    example = None
+
+    section = cs_container(
+        classes=["cs_container"], ids=["cs_{}_container".format(title)]
+    )
+
+    section += cs_header(classes=["cs_header"], text=parent_title)
+
+    section_content = cs_content(
+        classes=["cs_content"], ids=["cs_{}_content".format(title)]
+    )
+
+    section_content += nodes.paragraph(text=schema.get(MK.Description, ""))
+
+    for config_key in [MK.ElementValidators, MK.ContextValidators]:
+        element_nodes = [
+            nodes.paragraph(text=validator.msg)
+            for validator in schema.get(config_key, [])
+        ]
+        section_content.extend(element_nodes)
+
+    for config_key in [MK.LayerTransformation, MK.Transformation]:
+        if config_key in schema:
+            section_content += nodes.paragraph(text=schema.get(config_key).msg)
+
+    section_children = cs_children(
+        classes=["cs_children"], ids=["cs_{}_children".format(title)]
+    )
+
+    if schema[MK.Type] == types.NamedDict:
+        example = dict()
+        for key, value in schema[MK.Content].items():
+            generate(value, section_children, key)
+
+        for child in section_children.children:
+            example[child["parent"]] = child["example"]
+
+    elif schema[MK.Type] == types.List:
+        generate(
+            schema[MK.Content][MK.Item], section_children, parent_title="List Item"
+        )
+        example = [section_children.children[0]["example"]]
+
+    elif schema[MK.Type] == types.Dict:
+        example = dict()
+        generate(schema[MK.Content][MK.Key], section_children, parent_title="Key")
+        generate(schema[MK.Content][MK.Value], section_children, parent_title="Value")
+        example[section_children.children[0]["example"]] = section_children.children[1][
+            "example"
+        ]
+
+    elif isinstance(schema[MK.Type], types.BasicType):
+        example = "{}".format(BASIC_EXAMPLE.get(schema[MK.Type].name, "undefined"))
+        section_children += nodes.paragraph(
+            text=":type: {_type}".format(_type=schema[MK.Type].name)
+        )
+
+    else:
+        err_msg = "Unexpected type ({}) in schema while generating documentation."
+        raise TypeError(err_msg.format(schema[MK.Type]))
+
+    section_content += section_children
+
+    section += section_content
+    section["example"] = example
```

Comment:
I am not sure how robust this implementation is, adding an attribute to the `section` and then in the node later on, verify if it exists. But this was the simplest way of providing a mean to extract the info and put it into html code blocks. Note that if we provide latex etc as well, we can't do any language specific stuff here.

Reply:
For now this will be the method. Further investigation did not reveal a better way to do it.